server/routes/posts.js

```diff
@@ -13,7 +18,7 @@ router.get('/posts', async (req, res) => {
 });
 
 // Get all posts from a specific user
-router.get('/posts/user/:userId', async (req, res) => {
+router.get('/:userId', async (req, res) => {
```

Comment:
This route should technically be /user/:userId

so the full route would be /post/user/:userId, since we're specifying that this is a user ID and not a post ID